strictNullChecks with RequestOptions not passing

**I'm submitting a ...**  (check one with "x")
```
[X] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
With strictNullChecks turned on, I get this error:
```
ERROR in <xxx>\node_modules\@angular\http\src\base_request_options.d.ts
(63,18): error TS2459: Type 'RequestOptionsArgs | undefined' has no property 'method' and no string index signature.
```
It has the same error for method, headers, body, url, search, params, withCredentials, and responseType.

**Expected behavior**
The build passes without errors.

**Minimal reproduction of the problem with instructions**
```
let options: RequestOptionsArgs = { body: {} };
this.http.delete(url, new RequestOptions(options));
```

**What is the motivation / use case for changing the behavior?**
One of 4.1's features is passing strictNullChecks

**Please tell us about your environment:**
Typescript 2.3.2
tsconfig.json
```
 "compilerOptions": {
		"alwaysStrict": true,
		"emitDecoratorMetadata": true,
		"experimentalDecorators": true,
		"forceConsistentCasingInFileNames": true,
		"lib": [
			"es2015",
			"dom"
		],
		"module": "commonjs",
		"moduleResolution": "node",
		"noImplicitReturns": true,
		"sourceMap": true,
		"strictNullChecks": true,
		"suppressImplicitAnyIndexErrors": true,
		"target": "es5"
	}
```
* **Angular version:** 4.1.1
 
* **Language:** TypeScript 2.3.2

--- a/packages/http/src/base_request_options.ts
+++ b/packages/http/src/base_request_options.ts
@@ -79,9 +79,8 @@ export class RequestOptions {
   responseType: ResponseContentType|null;
 
   // TODO(Dzmitry): remove search when this.search is removed
-  constructor(
-      {method, headers, body, url, search, params, withCredentials,
-       responseType}: RequestOptionsArgs = {}) {
+  constructor(opts: RequestOptionsArgs = {}) {
+    const {method, headers, body, url, search, params, withCredentials, responseType} = opts;
     this.method = method != null ? normalizeMethodName(method) : null;
     this.headers = headers != null ? headers : null;
     this.body = body != null ? body : null;

--- a/packages/http/src/base_response_options.ts
+++ b/packages/http/src/base_response_options.ts
@@ -65,7 +65,8 @@ export class ResponseOptions {
    */
   type: ResponseType|null;
   url: string|null;
-  constructor({body, status, headers, statusText, type, url}: ResponseOptionsArgs = {}) {
+  constructor(opts: ResponseOptionsArgs = {}) {
+    const {body, status, headers, statusText, type, url} = opts;
     this.body = body != null ? body : null;
     this.status = status != null ? status : null;
     this.headers = headers != null ? headers : null;

--- a/tools/public_api_guard/http/http.d.ts
+++ b/tools/public_api_guard/http/http.d.ts
@@ -144,7 +144,7 @@ export declare class RequestOptions {
     /** @deprecated */ search: URLSearchParams;
     url: string | null;
     withCredentials: boolean | null;
-    constructor({method, headers, body, url, search, params, withCredentials, responseType}?: RequestOptionsArgs);
+    constructor(opts?: RequestOptionsArgs);
     merge(options?: RequestOptionsArgs): RequestOptions;
 }
 
@@ -192,7 +192,7 @@ export declare class ResponseOptions {
     headers: Headers | null;
     status: number | null;
     url: string | null;
-    constructor({body, status, headers, statusText, type, url}?: ResponseOptionsArgs);
+    constructor(opts?: ResponseOptionsArgs);
     merge(options?: ResponseOptionsArgs): ResponseOptions;
 }
 

